src/infrastructure/apiRequest.ts

```diff
@@ -108,7 +108,7 @@ async function apiRequest(requestConfig: RequestConfig, cacheTTL?: number): Prom
 
       // Add body for non-GET requests
       if (requestConfig.data && requestConfig.method && requestConfig.method.toUpperCase() !== "GET") {
-        if (typeof requestConfig.data === "string") {
+        if (typeof requestConfig.data === "string" || requestConfig.data instanceof FormData) {
```

Comment:
- Avoid stringify on `FormData`
- `fetch` sets headers according to type of `body`, so it will detect `FormData`